src/gtk-4.0/widgets/_windows.scss

```diff
@@ -4,71 +4,67 @@
     color: $fg-color;
 }
 
-window.csd {
-    margin: rem(12px);
-    border-radius: rem(6px);
-    box-shadow:
-        // Intentionally not in ems since it's used as a stroke
-        0 0 0 1px $toplevel-border-color,
-        // Force shadows to be the same size to prevent jumpy resize transition
-        0 13px 16px 4px transparent,
-        shadow(4);
-
-    &:backdrop {
-        box-shadow:
-            0 0 0 1px $toplevel-border-color,
-            0 13px 16px 4px transparent,
-            shadow(2);
-    }
-
-    &.maximized,
-    &.fullscreen,
-    &.tiled-top,
-    &.tiled-right,
-    &.tiled-bottom,
-    &.tiled-left {
-        margin: 0;
-    }
-
-    // black behind corners
-    &.maximized {
-        box-shadow: 0 0 0 rem(3px) black;
-    }
+window {
+    // Stolen from Adwaita, "lamefun trick to get rounded borders regardless of CSD use"
+    border-width: 0;
```

Comment:
I suspect this is a GTK3-ism too. I never figured out what it is or does and just left it in there 😅

Reply:
Haha yeah I wasn't sure since David's screenshots from before look really odd with like spacing around the headerbar